api/compose/v1beta2/stack.go

```diff
@@ -54,7 +54,7 @@ func (s *Stack) clone() *Stack {
 	result := new(Stack)
 	result.TypeMeta = s.TypeMeta
 	result.ObjectMeta = s.ObjectMeta
-	result.Spec = s.Spec.DeepCopy()
+	result.Spec = s.Spec.clone()
```

Comment:
Don't do that. This actually do not do a deep copy, and it has strong implications on how the API server detect actual changes in the stack spec.